### PR TITLE
Retheme chats

### PR DESCRIPTION
**Pull request by schancel**

### CODE REVIEW

**Review by maff1989 — COMMENTED** [review]
Line 66 of Chat.vue can be removed. I tested more and the q-banner doesn't even display. Will need to iterate over that at a later date.

File: src/cashweb/relay/index.ts [hunk 1]
```diff
@@ -520,6 +520,9 @@ export class RelayClient extends ReadOnlyRelayClient {
         if (item.type === 'text') {
           return this.messageConstructor.constructTextEntry({ ...item })
         }
+        if (item.type === 'reply') {
```
Comment (schancel) [inline src/cashweb/relay/index.ts:523]: Wot, when was this broken.

File: src/components/chat/ChatInput.vue [hunk 2]
```diff
@@ -1,59 +1,20 @@
...
-                  input-class="text-right"
-                />
-              </q-item-section>
-            </q-item>
-          </q-list>
-        </q-menu>
-      </q-btn>
-
-      <!-- <q-separator vertical /> -->
+      <q-btn
```
Comment (schancel) [inline src/components/chat/ChatInput.vue:4]: Let's do this in another PR.

File: src/components/chat/ChatList.vue [hunk 3]
```diff
@@ -19,6 +24,7 @@
             :value-unread="formatBalance(contact.totalUnreadValue)"
             :num-unread="contact.totalUnreadMessages"
             :compact="compact"
+            @click="vuexSetActiveChat(contact.address)"
```
Comment (schancel) [inline src/components/chat/ChatList.vue:27]: Why did this need to change?

Comment (schancel) [reply src/components/chat/ChatList.vue:27]: Does seem like a better place to put it though.

File: src/store/modules/chats.ts [hunk 4]
```diff
@@ -582,10 +596,26 @@ const module: Module<State, unknown> = {
         const textItem: TextItem = (newMsg.items.find(
           item => item.type === 'text',
         ) as TextItem) ?? { text: '' }
+        const stealthItem: StealthItem = (newMsg.items.find(
```
Comment (schancel) [inline src/store/modules/chats.ts:599]: This is fixing notifications?

File: src/components/chat/ChatList.vue (hunk 3 above)
Comment (maff1989) [reply src/components/chat/ChatList.vue:27]: I added logic for the Forum route (and potentially other, future, non-chat routes) wherein the `state.activeChatAddr` is set to `undefined`. With this logic, it made sense to move the vuex action to ChatList vs ChatListItem. That way I could set the `state.activeChatAddr` value of `undefined` for Forum, and also set the `contact.address` value within the same component.

File: src/router/index.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ const protectedRoutes = ['/forum/new-post']
 async function addContactFromNavigation<S>(store: Store<S>, address: string) {
   try {
     // Validate address
-    const apiAddress = toAPIAddress(address) // TODO: Make generic
+    const apiAddress = toDisplayAddress(address) // TODO: Make generic
```
Comment (schancel) [inline src/router/index.ts:22]: This seems very wrong.

File: src/store/modules/chats.ts [hunk 6]
```diff
@@ -340,16 +351,16 @@ const module: Module<State, unknown> = {
       }
     },
     setActiveChat(state, address) {
```
Comment (schancel) [inline src/store/modules/chats.ts:353]:
```
    setActiveChat(state, address) {
      state.activeChatAddr = address

      // make sure address is defined, e.g. Forum is undefined
      if (!address) {
        return;
      }

      const apiAddress = toDisplayAddress(address)

      if (!(apiAddress in state.chats)) {
        state.chats[apiAddress] = {
          ...defaultContactObject,
          messages: [],
          address: apiAddress,
        }
      }
    },
```

File: src/components/chat/messages/ChatMessage.vue [hunk 7]
```diff
@@ -18,88 +17,95 @@
     <q-dialog v-model="deleteDialog">
       <delete-message-dialog
         :address="address"
-        :payload-digest="message.payloadDigest"
+        :payload-digest="payloadDigest"
```
Comment (schancel) [inline src/components/chat/messages/ChatMessage.vue:20]: Is this a bug on master?

File: src/css/app.scss [hunk 8]
```diff
@@ -4,6 +4,7 @@
 
 :root {
   --q-on-primary-color-text: white;
+  font-size: large;
```
Comment (schancel) [inline src/css/app.scss:7]: Do we need this change?
